flight/Modules/Actuator/actuator.c

```diff
@@ -218,282 +231,416 @@ static float get_curve2_source(ActuatorDesiredData *desired, SystemSettingsAirfr
 	return 0;
 }
 
-/**
- * @brief Main Actuator module task
- *
- * Universal matrix based mixer for VTOL, helis and fixed wing.
- * Converts desired roll,pitch,yaw and throttle to servo/ESC outputs.
- *
- * Because of how the Throttle ranges from 0 to 1, the motors should too!
- *
- * Note this code depends on the UAVObjects for the mixers being all being the same
- * and in sequence. If you change the object definition, make sure you check the code!
- *
- * @return -1 if error, 0 if success
- */
-static void actuator_task(void* parameters)
+static void compute_one_mixer(int i,
+		int16_t (*vals)[MIXERSETTINGS_MIXER1VECTOR_NUMELEM],
+		MixerSettingsMixer1TypeOptions type)
 {
-	float dT = 0.0f;
+	types_mixer[i] = type;
 
+	i *= MIXERSETTINGS_MIXER1VECTOR_NUMELEM;
+
+	if ((type != MIXERSETTINGS_MIXER1TYPE_SERVO) &&
+			(type != MIXERSETTINGS_MIXER1TYPE_MOTOR)) {
+		for (int j = 0; j < MIXERSETTINGS_MIXER1VECTOR_NUMELEM; j++) {
+			// Ensure unused types are zero-filled
+			motor_mixer[i+j] = 0;
+		}
+	}
+
+	for (int j = 0; j < MIXERSETTINGS_MIXER1VECTOR_NUMELEM; j++) {
+		motor_mixer[i+j] = (*vals)[j] * (1.0f / MIXER_SCALE);
+	}
```

Comment:
Do you even need to run this for-loop when a mixer is set disabled? The values are zero anyway (as per UAVO browser), also as ensured above.